core/controllers/editor.py

```diff
@@ -175,14 +175,23 @@ def get(self, exploration_id):
         # are not used by that tab.
         version = self.request.get('v', default_value=None)
         apply_draft = self.request.get('apply_draft', default_value=False)
+
+        user_settings = user_services.get_user_settings(self.user_id)
```

Comment:
Is it possible for self.user_id to be None?

Reply:
yes it can be. I added an if clause to assign the values of has_seen_editor_tutorial and has_seen_translation_tutorial only if there is an user logged in.

Reply:
There should be tests to ensure correct behaviour in the None and non-None cases.

Reply:
Yes I will be adding tests for editor.py (hopefully tomorrow). Can I take this up in that PR, since it will keep things organised and easy to track?

Reply:
In general, any PR submitted to the codebase must come with full test coverage for any new code added in that PR. I think we should probably start following that philosophy from the get-go.

Reply:
Done